### ISSUE TITLE
AJAX: OnChangeAjaxBehavior and AjaxFormComponentUpdatingBehavior fail when replacing component

### ISSUE THREAD

**Issue by kbachl** [brix-cms/brix-cms#210]
ERROR - DefaultExceptionMapper     - Unexpected error occurred
org.apache.wicket.WicketRuntimeException: No Page found for component [Component id = history]
	at org.apache.wicket.Component.getPage(Component.java:1755)
	at org.apache.wicket.ajax.AjaxRequestHandler.add(AjaxRequestHandler.java:231)
	at org.brixcms.demo.web.tile.ajax.AjaxDemoPanel$2.onUpdate(AjaxDemoPanel.java:123)....

in case one uses a code like e.g.:

OnChangeAjaxBehavior onChangeAjaxBehavior = new OnChangeAjaxBehavior() {
            @Override
            protected void onUpdate(AjaxRequestTarget target) {
                Label c = new Label("history", "label");
                c.setOutputMarkupId(true);
                target.add(c);
                addOrReplace(c);
            }
        };
        field.add(onChangeAjaxBehavior);

as found in the wicket examples

**Comment by kbachl** [comment]
@martin-g can you please shed some light for me as I dont understand why we get a No Page found error in wicket 8 - the classic AutoCompleteTextField will work however; I suppose this is due to not adding a new component to the hirarchy but using the old one instead;

I add a pull request with a new ajax demo tile that can be fast changed to this mess...

**Comment by kbachl** [comment]
PS: it doesnt matter if the page is the root / using index.brix or any other brix page - so the error is somwhere deeper in it I guess

**Comment by kbachl** [comment]
#211 is the request

**Comment by kbachl** [comment]
update: is is an wicket8 error! - I will make a jira with quickstart for it

**Comment by kbachl** [comment]
https://issues.apache.org/jira/browse/WICKET-6415

**Comment by kbachl** [comment]
it is resolved - problem was the order of the ajax call - error warning of wicket got enhanced by andrea del bene

**Comment by kbachl** [comment]
PS: the example will be added to brix-demo
